examples/process.rs

```diff
@@ -0,0 +1,373 @@
+    let output_str =
+      v8::String::new(&mut self_.context_scope, "output").unwrap();
+    self_.context.global(&mut self_.context_scope).set(
+      &mut self_.context_scope,
+      output_str.into(),
+      output.into(),
+    );
+
+    // execute script
+    self_.execute_script(source);
+
+    let process_str =
+      v8::String::new(&mut self_.context_scope, "Process").unwrap();
+    let process_fn = self_
+      .context
+      .global(&mut self_.context_scope)
+      .get(&mut self_.context_scope, process_str.into())
+      .expect("missing function Process");
+
+    let process_fn = unsafe { v8::Local::<'_, v8::Function>::cast(process_fn) };
+    self_.process_fn = Some(process_fn);
+
+    self_
+  }
+
+  fn execute_script(&mut self, script: v8::Local<'s, v8::String>) {
+    let scope = &mut v8::HandleScope::new(&mut self.context_scope);
+    let try_catch = &mut v8::TryCatch::new(scope);
+
+    let script = v8::Script::compile(try_catch, script, None)
+      .expect("failed to compile script");
+
+    if script.run(try_catch).is_none() {
+      let exception = try_catch.exception().unwrap();
+      let exception_string = exception
+        .to_string(try_catch)
+        .unwrap()
+        .to_rust_string_lossy(try_catch);
+
+      panic!("{}", exception_string);
+    }
+  }
+
+  /// Processes the given HTTP request.
+  pub fn process<R>(&mut self, request: R)
+  where
+    R: HttpRequest + 'static,
+  {
+    let request: Box<dyn HttpRequest> = Box::new(request);
+    let request = self.wrap_request(request);
+
+    let scope = &mut v8::HandleScope::new(&mut self.context_scope);
+    let try_catch = &mut v8::TryCatch::new(scope);
+
+    let process_fn = self.process_fn.as_mut().unwrap();
+    let global = self.context.global(try_catch).into();
+
+    if process_fn
+      .call(try_catch, global, &[request.into()])
+      .is_none()
+    {
+      let exception = try_catch.exception().unwrap();
+      let exception_string = exception
+        .to_string(try_catch)
+        .unwrap()
+        .to_rust_string_lossy(try_catch);
+
+      panic!("{}", exception_string);
+    }
+  }
+
+  /// Utility function that wraps a http request object in a JavaScript object.
+  fn wrap_request(
+    &mut self,
+    request: Box<dyn HttpRequest>,
+  ) -> v8::Local<'s, v8::Object> {
+    // TODO: support ObjectTemplate (SetAccessor is not supported?)
+    // TODO: use internal field (SetInternalField is not supported?)
+    // TODO: fix memory leak
+
+    use std::ffi::c_void;
+
+    // Dobule-box to get C-sized reference of Box<dyn HttpRequest>
+    let request = Box::new(request);
+
+    // Local scope for temporary handles.
+    let scope = &mut self.context_scope;
+
+    let result = v8::Object::new(scope);
+    let external = v8::External::new(
+      scope,
+      Box::leak(request) as *mut Box<dyn HttpRequest> as *mut c_void,
+    );
+
+    let key = v8::String::new(scope, "__private__").unwrap().into();
```

Comment:
Oh, we don't have a binding for `v8::Object::SetInternalField()` yet, do we? And no way to instantiate a `v8::Private` either. I've opened #477 for the former.

Reply:
Thanks!

Reply:
#477 landed, if you want to update.

Reply:
Now it uses `ObjectTemplate` and internal fields.